closed/autoconf/custom-hook.m4

```diff
@@ -342,16 +342,26 @@ AC_DEFUN([OPENJ9_PLATFORM_SETUP],
   AC_ARG_WITH(noncompressedrefs, [AS_HELP_STRING([--with-noncompressedrefs],
     [build non-compressedrefs vm (large heap)])])
 
+  AC_ARG_WITH(mixedrefs, [AS_HELP_STRING([--with-mixedrefs],
+    [build mixedrefs vm ])])
+
   # When compiling natively host_cpu and build_cpu are the same. But when
   # cross compiling we need to work with the host_cpu (which is where the final
   # JVM will run).
   OPENJ9_PLATFORM_EXTRACT_VARS_FROM_CPU($host_cpu)
 
-  if test "x$with_noncompressedrefs" = x ; then
-    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}_cmprssptrs"
-    OPENJ9_LIBS_SUBDIR=compressedrefs
+  OPENJ9_ENABLE_MIXED_REFERENCES=false
+  if test "x$with_mixedrefs" = xno -o "x$with_mixedrefs" = x; then
+    if test "x$with_noncompressedrefs" = xno -o "x$with_noncompressedrefs" = x ; then
+      OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}_cmprssptrs"
+      OPENJ9_LIBS_SUBDIR=compressedrefs
+    else
+      OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}"
+      OPENJ9_LIBS_SUBDIR=default
+    fi
   else
-    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}"
+    OPENJ9_ENABLE_MIXED_REFERENCES=true
+    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}_mxdptrs"
     OPENJ9_LIBS_SUBDIR=default
   fi
```

Comment:
I think this reads better if you use `if` and `elif`:
```
  if test "x$with_mixedrefs" = xyes ; then
    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}_mxdptrs"
    OPENJ9_ENABLE_MIXED_REFERENCES=true
    OPENJ9_LIBS_SUBDIR=default
  elif test "x$with_noncompressedrefs" = xyes ; then
    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}_cmprssptrs"
    OPENJ9_ENABLE_MIXED_REFERENCES=false
    OPENJ9_LIBS_SUBDIR=compressedrefs
  else
    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_CPU}"
    OPENJ9_ENABLE_MIXED_REFERENCES=false
    OPENJ9_LIBS_SUBDIR=default
  fi
```
(one level of nesting, setting variables in the same order)